### PR TITLE
Add prettier and run on codebase

### PR DESCRIPTION
**Pull request by jonathaningram**
As I promised in #1229, this PR adds support for prettier but does not run it across the codebase. If this gets approved I can either run it across the codebase in a subsequent commit.

I really hope this gets accepted! 🤞 In my env var task, my dev workflow was hindered because I couldn't just hit `cmd+s` and go on my way - I had to deal with eslint complaining that I had wrong spacing here and there and had to manually fix those issues. By the way, the code that we pushed in my env var PR was actually all prettier formatted, I just didn't have the "format on save" enabled for the project.

Notes and comments:

1. `.prettierrc.json`: normally I'd not even have this file at all. The only reason I have it is so that I can override the `singleQuote` setting to match Airbnb. However, like `gofmt`, the point is that you don't need to configure it - you just format and forget. I'm super happy and _encourage_ us to remove this entirely and only use Airbnb as a base for non-style things. If you do approve this PR, please advise if you're happy for me to remove this option override (like `gofmt`, I just want to format things one way without thinking about having this particular option enabled for this particular project).
2. You can see how we get formatting for CSS, JS, JSX and JSON all "out of the box" - basically our entire frontend stack can now be auto-formatted.
3. We can format `.eslintrc` too (it's just JSON). In a separate PR I'll look to upgrade eslint and associated packages. I'd do this PR because eslint now recommends to use `.eslintrc.json` as the filename. And then once we do that, it will be another JSON file in the repo that will "magically" just become auto-formatted.
4. There are some unrelated spelling error fixes in a separate commit in this PR...

Look forward to hearing back and thanks! 😎

### CODE REVIEW

**Comment by jonathaningram** [comment]
Thanks @jcscottiii awesome!!! I'll leave the actual running of the formatting until my other ones are merged so I can avoid conflicts.

**Comment by jonathaningram** [comment]
@jcscottiii this is rebased and ready to merge. Will be no more conflicts unless you merge #1262 first, but it doesn't bother me which one is merged first.

Also note that prettier can now format markdown files too, hence the changes in this PR.

Also worth noting is that I've disabled a bunch of eslint rules for now as we transition and upgrade airbnb/javascript.

File: .eslintrc [hunk 1]
```diff
@@ -8,12 +8,35 @@
...
+    "global-require": 0,
+    "import/imports-first": 0,
+    "import/newline-after-import": 0,
+    "import/no-extraneous-dependencies": 0,
+    "import/no-named-as-default": 0,
+    "import/no-unresolved": [2, { "ignore": ["skin"] }],
+    "import/prefer-default-export": 0,
+    "jsx-a11y/href-no-hash": 0,
+    "jsx-a11y/img-has-alt": 0,
+    "no-console": 0,
```
Comment (jcscottiii) [inline .eslintrc]: some of these are fine during the transition but ones like this one `no-console` and ones like `prefer-const` `no-undef` seem like we should never have to turn these off.

**Comment by jcscottiii** [comment]
@jonathaningram i think i need a better understanding in terms of what the path to re-enabling these rules are. there's quite a few.

also, is it just possible to use https://github.com/prettier/eslint-config-prettier which will stop the conflict between prettier and eslint without us having to manually turn off rules?

File: .eslintrc (hunk 1 above)
Comment (jonathaningram) [reply .eslintrc]: Yep we never want to turn those off. In other words, we should always prefer const, for example.

I have grouped all of the ones we need to revisit after my TODO comment.

**Comment by jonathaningram** [comment]
> @jonathaningram i think i need a better understanding in terms of what the path to re-enabling these rules are. there's quite a few.
> 
> also, is it just possible to use https://github.com/prettier/eslint-config-prettier which will stop the conflict between prettier and eslint without us having to manually turn off rules?

@jcscottiii I think the most obvious path after this is merged is to just go through each rule one by one. Re-enable the rule, run the linter and fix the issues, then provide a PR. If a rule only results in one error maybe it's worth fixing more than one in that PR.

I could have tried to fix all of these in this PR, but I didn't want it to drag out too long so that's why I decided to disable the rules as part of a transition period. I'm pretty sure ones like `prefer-const` can even be fixed by running `eslint --fix` so some changes should be quick and automatic.

And yup, we're using `eslint-config-prettier` 😄  that is what is silencing all of the airbnb formatting rules. To be clear, the reason we need to silent these extra rules is that we _had_ to upgrade to a newer version of airbnb (not the latest one available yet, just a newer one). We had to upgrade because of the few eslint configs and plugins that also needed to update. So it was about finding a bunch of deps that would work together.

**Comment by jonathaningram** [comment]
@jcscottiii ok have rebased this since the latest merge.

**Comment by jonathaningram** [comment]
@jcscottiii just checking that you're not waiting on anything from me to merge this?

**Comment by jcscottiii** [comment]
@jonathaningram I appreciate the work going into this PR. unfortunately, I don't want to merge it as-is because if we disable rules now and slowly turn them back on in later PRs, in between then, other developers (e.g. we have a new developer coming on our team) might come and start contributing bad code because of those turned off rules.

**Comment by jonathaningram** [comment]
@jcscottiii I'm happy to disable the rules and fix the code that fails those rules, it just means the PR will be bigger and viewing changes might be hard. Having said that, if I fix the code that fails those rules in a separate commit in this PR you should be able to just view that single commit's changes.

Alternatively, I'm able to do a PR almost immediately after this is merged, so there'll really only be a small time in between. Up to you of course.

**Comment by jcscottiii** [comment]
@jonathaningram  that's a good point. if it will be too bloated, create a bunch of smaller PRs that use this one as the base for the first one (first PR will use this as base, the PR after that will use the first PR as base, and so on to create a long chain of PRs. also it can be many chains if the changes aren't that dependent on one another) until all of those rules are re-enabled then I can review and merge them

**Comment by jonathaningram** [comment]
@jcscottiii ok I decided to go ahead and just fix as many lint issues as I could. You'll notice I've still got some overrides in place. As follows:

```json
{
  "import/imports-first": 0,
  "import/no-extraneous-dependencies": 0,
  "import/no-named-as-default": 0,
  "react/jsx-filename-extension": 0
}
```

And test eslint file:

```json
{
  "jasmine/new-line-before-expect": 0,
  "jasmine/new-line-between-declarations": 0
}
```

1. `import/no-extraneous-dependencies`: it's because we are importing `skin` and `dashboard` modules that don't exist in package.json. This eslint rule doesn't have the feature to ignore project-level modules so we have no choice but to turn this off for now.
2. `import/no-named-as-default`: a bit difficult to fix in the codebase for this PR. The current codebase is already failing this test anyway, so I don't think it needs to be done in this PR.
3. `react/jsx-filename-extension`: we'll need to revisit and rename files accordingly. Note: pretty sure later airbnb versions have turned this off, so I don't want to make any changes on this right now, especially renaming all the files...
4. The 3 other rules I've turned off are just formatting so we can do this in a subsequent PR. 

Thanks for you time :)

**Comment by jonathaningram** [comment]
@jcscottiii I've just rebased again after my previous merge.

**Comment by jonathaningram** [comment]
@pburkholder I've rebased and Circle is passing. Are you able to merge this without Codeclimate passing? I don't think this was in place when I started this PR, plus this PR doesn't really change any code, it' just formatting it - I didn't want to go down a rabbit whole massive PR that refactored all the code :)

**Comment by pburkholder** [comment]
Let's make the security-related checks go away, then we'll address the PR on its own merits.

And yes, this wasn't in place before.

**Comment by pburkholder** [comment]
If you cherry-pick 47387115acd64535dacd135764c17c2b7664fb3f then this should be mergeable and addresses the CodeClimate security findings.

**Comment by jonathaningram** [comment]
@pburkholder yep I'll look into it. Looks like a few commits/changes have gone into master related to package.json/npm-shrinkwrap.json so the rebase might be a bit intensive, but stay tuned.

**Comment by pburkholder** [comment]
Thanks for the help Jonathan. We had to turn on CodeClimate, but it looks like we're really close to getting this in. I won't have time to work on this today, if you can take a stab at it, that would be great. P.

**Comment by pburkholder** [comment]
Hi Jonathan -- how can I help get this across the finish line? Happy to pair if you want to.

**Comment by jonathaningram** [comment]
@pburkholder I believe this is ready. There are still 27 CC issues but they are just odd rules like too many returns, not security issues. Let me know if I missed something?

**Comment by jonathaningram** [comment]
@pburkholder yay!

**Pull request merged**